src/main/java/com/google/devtools/build/lib/authandtls/GoogleAuthCallCredentialsProvider.java

```diff
@@ -0,0 +1,44 @@
+package com.google.devtools.build.lib.authandtls;
+
+import com.google.auth.Credentials;
+import com.google.common.base.Preconditions;
+import com.google.devtools.build.lib.concurrent.ThreadSafety.ThreadSafe;
+import io.grpc.CallCredentials;
+import io.grpc.auth.MoreCallCredentials;
+import java.io.IOException;
+import java.util.concurrent.TimeUnit;
+
+@ThreadSafe
+public class GoogleAuthCallCredentialsProvider implements CallCredentialsProvider {
+
+  private final Credentials credentials;
+
+  private CallCredentials callCredentials;
+  private long lastRefreshTime;
+
+  public GoogleAuthCallCredentialsProvider(Credentials credentials) {
+    Preconditions.checkNotNull(credentials, "credentials");
+    this.credentials = credentials;
+    callCredentials = MoreCallCredentials.from(credentials);
+  }
+
+  @Override
+  public CallCredentials getCallCredentials() {
+    return callCredentials;
+  }
+
+  @Override
+  public void refresh() throws IOException {
+    synchronized (this) {
```

Comment:
Upstream is already thread-safe, so this isn't needed.

Reply:
IIUC, we need this guard for `lastRefreshTime`.